x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/persistence/TrainedModelProvider.java

```diff
@@ -117,6 +120,30 @@ public void getTrainedModel(String modelId, ActionListener<TrainedModelConfig> l
                 listener::onFailure));
     }
 
+    public void deleteTrainedModel(String modelId, ActionListener<Boolean> listener) {
+        DeleteByQueryRequest request = new DeleteByQueryRequest().setAbortOnVersionConflict(false);
+
+        request.indices(InferenceIndexConstants.INDEX_PATTERN);
+        QueryBuilder query = QueryBuilders.termQuery(TrainedModelConfig.MODEL_ID.getPreferredName(), modelId);
+        request.setQuery(query);
+        request.setRefresh(true);
+
+        executeAsyncWithOrigin(client, ML_ORIGIN, DeleteByQueryAction.INSTANCE, request, ActionListener.wrap(deleteResponse -> {
```

Comment:
Maybe not directly related to this PR, but what is the usual practice in ES wrt idempotency? Should exception be thrown when the entity being deleted does not exist?

Reply:
For the rest of ML we do throw "not found" errors if you delete something that doesn't exist.  But we don't throw errors for stopping a stopped thing or closing a closed thing.  So what's here is consistent with that.

Reply:
Thanks for explanation!